### PR TITLE
[in progress] Create 2 executable for bolt and add ros2 launch

### PR DESCRIPTION
**Pull request by Lhumd**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

Add safe mode and make it able to run with single process.

## Description

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

Add safe mode, 
## How I Tested
By testing it on robot.

[//]: # "Explain how you tested your changes"

## Do not merge before

[//]: # "Link other open pull request here that need to be merged first."
[//]: # "Remove this section if it does not apply."

- [ ] link to dependent pull request

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [ ] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [ ] All new functions/classes are documented and existing documentation is updated according to changes.
- [ ] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: include/bolt/dgm_bolt.hpp [hunk 1]
```diff
@@ -73,6 +77,13 @@ class DGMBolt : public dynamic_graph_manager::DynamicGraphManager
         mim_msgs::srv::JointCalibration::Request::SharedPtr req,
         mim_msgs::srv::JointCalibration::Response::SharedPtr res);
 
+    /**
```
Comment (MaximilienNaveau) [inline include/bolt/dgm_bolt.hpp]: this is a copy of the documentation above, to be removed

File: launch/bolt_bringup.launch.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from launch import LaunchDescription
+from launch_ros.actions import Node
+
+
+def generate_launch_description():
+    return LaunchDescription(
+        [
+            Node(
```
Comment (MaximilienNaveau) [inline launch/bolt_bringup.launch.py]: nice

File: launch/bolt_bringup.launch.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                package="bolt",
+                node_executable="dg",
+                output="screen",
+                prefix=['xterm -fa "Monospace" -fs 14 -hold -e'],
+                # prefix=['xterm -e gdb -ex=r --args'],
+            ),
+            Node(
+                package="bolt",
```
Comment (MaximilienNaveau) [inline launch/bolt_bringup.launch.py]: won't work as you need sudo

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -84,6 +84,54 @@ if(${dynamic_graph_manager_FOUND})
   # Export.
   list(APPEND all_exe_targets dg_main_bolt)
 
+  #
+  # Create the robot executable.
+  #
+  add_executable(dg programs/dg.cpp)
```
Comment (MaximilienNaveau) [inline src/CMakeLists.txt]: target names needs more verbosity, at least the package name:
bolt_dg
bolt_hw

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -84,6 +84,54 @@ if(${dynamic_graph_manager_FOUND})
...
+    dg
+    PUBLIC DYNAMIC_GRAPH_MANAGER_YAML_PATH="${dynamic_graph_manager_yaml_file}"
+           ODRI_CONTROL_INTERFACE_YAML_PATH="${odri_control_interface_yaml_path}")
+  # Export.
+  list(APPEND all_exe_targets dg)
+
+  #
+  # Create the robot executable.
+  #
+  add_executable(hw programs/hw.cpp)
```
Comment (MaximilienNaveau) [inline src/CMakeLists.txt]: see previous comment

File: src/programs/hw.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
+/**
+ * \file bolt.cpp
+ * \brief Execute the main program to control the bolt
+ * \author Maximilien Naveau
+ * \date 2018
+ *
+ * DynamicGraphManager for bolt main executbale.
```
Comment (MaximilienNaveau) [inline src/programs/hw.cpp]: typo

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]
First review done

File: src/dgm_bolt.cpp [hunk 7]
```diff
@@ -44,6 +44,7 @@ bool DGMBolt::is_in_safety_mode()
       }
       counter++;
     }
+    return false;
```
Comment (MaximilienNaveau) [inline src/dgm_bolt.cpp]: ... dangerous!!!!

File: src/CMakeLists.txt [hunk 8]
```diff
@@ -20,7 +20,7 @@ target_link_libraries(bolt Eigen3::Eigen)
 string(
   CONCAT odri_control_interface_yaml_path
          "${PythonModules_robot_properties_bolt_PATH}/"
-         "robot_properties_bolt/robot_properties_bolt/odri_control_interface/"
+         "robot_properties_bolt/resources/odri_control_interface/"
```
Comment (MaximilienNaveau) [inline src/CMakeLists.txt:23]: Good point!

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]

File: launch/bolt_bringup.launch.py (hunk 3 above)
Comment (Lhumd) [reply launch/bolt_bringup.launch.py]: Then I remove the bolt bringup.launch.py

Comment (MaximilienNaveau) [reply launch/bolt_bringup.launch.py]: That's what I did, I run this node by hand and then use a ros2 launch for the rest of the application

**Pull request merged**